lib/elixir/lib/map.ex

```diff
@@ -211,13 +220,16 @@ defmodule Map do
   """
   @spec take(map, [key]) :: map
   def take(map, keys) do
-    Enum.reduce(keys, [], fn key, acc ->
-      case fetch(map, key) do
-        {:ok, value} -> [{key, value} | acc]
-        :error -> acc
-      end
-    end)
-    |> :maps.from_list
+    do_take(keys, map, [])
+  end
```

Comment:
I should note that I'm not gonna do the guard clause for lists here, `Enum.to_list` is a plenty fast enough way to handle both cases. `:lists.reverse` that one extra just gets lost in the noise